File: apps/api/app/services/twiml.py
"""Small TwiML response helpers for WhatsApp replies."""

from html import escape

from app.models.task import ExtractedTask


def confirmation_reply(task: ExtractedTask) -> str:
    date_part = f" - {task.due_date}" if task.due_date else ""
    return twiml_message(f"Registre: {task.task_title}{date_part}. Confirmas?")


def clarification_reply() -> str:
    return twiml_message("No entendi bien. Me podes repetir que hay que hacer y para cuando?")


def unregistered_sender_reply() -> str:
    return twiml_message("No te tengo registrado para esta organizacion. Pedi acceso a coordinacion.")


def twiml_message(body: str) -> str:
    return f"<Response><Message>{escape(body)}</Message></Response>"

File: src/services/twiml.ts
import { ExtractedTask } from '../types';

export function confirmationReply(task: ExtractedTask): string {
  const datePart = task.due_date ? ` — ${task.due_date}` : '';
  const body = `Registré: ${task.task_title}${datePart}. Confirmas?`;
  return twimlMessage(body);
}

export function clarificationReply(): string {
  return twimlMessage('No entendí bien. ¿Me podés repetir qué hay que hacer y para cuándo?');
}

function twimlMessage(body: string): string {
  return `<Response><Message>${body}</Message></Response>`;
}
